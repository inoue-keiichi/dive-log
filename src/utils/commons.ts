export const getSimpleDate = (date: Date) => {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();
  return `${year}-${month < 10 ? "0" + month : month}-${
    day < 10 ? "0" + day : day
  }`;
};

const getURL = () => {
  let url =
    process.env.NEXT_PUBLIC_SITE_URL ?? // Set this to your site URL in production env.
    process.env.NEXT_PUBLIC_VERCEL_URL ?? // Automatically set by Vercel.
    "http://localhost:3000";
  // Make sure to include `https://` when not localhost.
  url = url.includes("http") ? url : `https://${url}`;
  // Make sure to exclude a trailing `/`.
  url = url.charAt(url.length - 1) !== "/" ? url : url.slice(0, -1);
  return url;
};

export const SITE_URL = getURL();
