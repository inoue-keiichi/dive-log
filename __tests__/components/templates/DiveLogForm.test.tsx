import DiveLogForm from "@/components/templates/DiveLogForm";
import { getSimpleDate } from "@/utils/commons";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import {
  fillAverageDepth,
  fillDate,
  fillDivingEndTime,
  fillDivingStartTime,
  fillMaxDepth,
  fillMemo,
  fillPlace,
  fillPoint,
  fillTankEndPressure,
  fillTankStartPressure,
  fillTemprature,
  fillTransparency,
  fillWaterTemprature,
  fillWeight,
} from "../__utils__/diveLogForm";
import renderer from "react-test-renderer";
import {
  click,
  selectIconRadioButton,
  selectRadioButton,
} from "../__utils__/common";

jest.setTimeout(10 * 1000);

// 新規のDiveLogFormを作ると現在の日付が初期値になる。
// 期待値を固定するためにモック化する。
const mockDate = new Date("2023/06/27");
jest.useFakeTimers();
jest.setSystemTime(mockDate);

const submit = jest.fn();
const back = jest.fn();
const deletee = jest.fn(); // 予約語

afterEach(() => {
  submit.mockClear();
  back.mockClear();
  deletee.mockClear();
});

describe("test", () => {
  test("succeeded in submitting form", async () => {
    render(<DiveLogForm onSubmit={submit} onBack={back} />);

    fillDate("2022-04-01");
    fillPlace("Ose");
    fillPoint("Wannai");
    fillDivingStartTime("09:30");
    fillDivingEndTime("10:00");
    fillAverageDepth("18");
    fillMaxDepth("25");
    fillTankStartPressure("200");
    fillTankEndPressure("50");
    click(screen.getByLabelText("スチール"));
    fillWeight("5");
    click(screen.getByLabelText("ウェット"));
    click(screen.getByLabelText("晴れ"));
    fillTemprature("35");
    fillWaterTemprature("28");
    fillTransparency("8");
    fillMemo("Good Diving!!");
    const target = screen.getByText("追加");
    fireEvent.click(target);

    // react-hook-form によって呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(1));
    expect(back).toHaveBeenCalledTimes(0);
    expect(submit.mock.calls[0][0]).toStrictEqual({
      date: "2022-04-01",
      place: "Ose",
      point: "Wannai",
      divingStartTime: "09:30",
      divingEndTime: "10:00",
      averageDepth: 18,
      maxDepth: 25,
      tankStartPressure: 200,
      tankEndPressure: 50,
      tankKind: "STEEL",
      weight: 5,
      suit: "WET",
      weather: "SUNNY",
      temprature: 35,
      waterTemprature: 28,
      transparency: 8,
      memo: "Good Diving!!",
    });
  });

  test("succeeded in submitting form without values", async () => {
    render(<DiveLogForm onSubmit={submit} onBack={back} />);
    const target = screen.getByText("追加");
    await fireEvent.click(target);
    // react-hook-form によって呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(1));
    expect(back).toHaveBeenCalledTimes(0);
    expect(submit.mock.calls[0][0]).toStrictEqual({
      date: getSimpleDate(new Date()),
      point: "",
      waterTemprature: undefined,
      transparency: undefined,
      averageDepth: undefined,
      divingEndTime: undefined,
      divingStartTime: undefined,
      maxDepth: undefined,
      memo: "",
      place: "",
      suit: undefined,
      tankEndPressure: undefined,
      tankKind: undefined,
      tankStartPressure: undefined,
      temprature: undefined,
      weather: undefined,
      weight: undefined,
    });
  });

  test("succeeded in submitting form with default values", async () => {
    const diveLog = {
      date: "2023-06-27",
      point: "Ose",
      waterTemprature: 28,
      transparency: 8,
    };

    render(<DiveLogForm diveLog={diveLog} onSubmit={submit} onBack={back} />);
    const target = screen.getByText("上書き");
    await fireEvent.click(target);
    // react-hook-form によって呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(1));
    expect(back).toHaveBeenCalledTimes(0);
  });

  test("succeeded in going back page", async () => {
    render(<DiveLogForm onSubmit={submit} onBack={back} />);

    const target = screen.getByTestId("back-button");
    await fireEvent.click(target);

    // react-hook-form によって呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(0));
    expect(back).toHaveBeenCalledTimes(1);
  });

  test("succeeded in delete button", async () => {
    const diveLog = {
      date: "2023-06-27",
      point: "Ose",
      waterTemprature: 28,
      transparency: 8,
    };

    render(
      <DiveLogForm
        diveLog={diveLog}
        onSubmit={submit}
        onBack={back}
        onDelete={deletee}
      />
    );

    const target = screen.getByTestId("dive-log-delete-button");
    await fireEvent.click(target);

    // react-hook-form によって呼び出されるまで待機
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(0));
    expect(back).toHaveBeenCalledTimes(0);
    expect(deletee).toHaveBeenCalledTimes(1);
  });
});

describe.skip("snapshot test", () => {
  test("new Dive log form", async () => {
    const tree = renderer
      .create(<DiveLogForm onSubmit={submit} onBack={back} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("existing Dive log form", async () => {
    const diveLog = {
      date: "2023-06-27",
      point: "Ose",
      waterTemprature: 28,
      transparency: 8,
      averageDepth: null,
      divingEndTime: "00:00",
      divingStartTime: "00:00",
      maxDepth: null,
      memo: "",
      place: "",
      suit: null,
      tankEndPressure: null,
      tankKind: null,
      tankStartPressure: null,
      temprature: null,
      weather: null,
      weight: null,
    };

    const tree = renderer
      .create(<DiveLogForm diveLog={diveLog} onSubmit={submit} onBack={back} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
