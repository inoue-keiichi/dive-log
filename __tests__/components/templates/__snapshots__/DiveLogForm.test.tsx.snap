// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test existing Dive log form 1`] = `
<form
  className="MuiStack-root css-1p5q5e5-MuiStack-root"
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-6ob1nr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="back-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIosNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="date"
    >
      日付
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        defaultValue="2023-06-27"
        disabled={false}
        id="date"
        name="date"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="date"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            日付
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="point"
    >
      ポイント
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        defaultValue="Ose"
        disabled={false}
        id="point"
        name="point"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            ポイント
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="waterTemprature"
    >
      水温
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        defaultValue={28}
        disabled={false}
        id="waterTemprature"
        name="waterTemprature"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          ℃
        </p>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            水温
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="transparency"
    >
      透明度
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        defaultValue={8}
        disabled={false}
        id="transparency"
        name="transparency"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          m
        </p>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            透明度
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    上書き
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</form>
`;

exports[`snapshot test new Dive log form 1`] = `
<form
  className="MuiStack-root css-1p5q5e5-MuiStack-root"
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-6ob1nr-MuiButtonBase-root-MuiIconButton-root"
        data-testid="back-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowBackIosNewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="date"
    >
      日付
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        defaultValue="2023-06-27"
        disabled={false}
        id="date"
        name="date"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="date"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-14lo706"
        >
          <span>
            日付
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="point"
    >
      ポイント
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="point"
        name="point"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            ポイント
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="waterTemprature"
    >
      水温
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="waterTemprature"
        name="waterTemprature"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          ℃
        </p>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            水温
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="transparency"
    >
      透明度
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="transparency"
        name="transparency"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          m
        </p>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            透明度
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      className="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    追加
  </button>
</form>
`;
